src/allmydata/test/no_network.py

```diff
@@ -258,7 +261,7 @@ def create_no_network_client(basedir):
     return defer.succeed(client)
 
 
-class _NoNetworkClient(_Client):
+class _NoNetworkClient(_Client):  # type: ignore # Cannot determine consistent MRO order
```

Comment:
Here's what I see:

```
tahoe-lafs 3399.mypy $ git diff
diff --git a/src/allmydata/test/no_network.py b/src/allmydata/test/no_network.py
index ba497f81a..17a63a3a2 100644
--- a/src/allmydata/test/no_network.py
+++ b/src/allmydata/test/no_network.py
@@ -261,7 +261,7 @@ def create_no_network_client(basedir):
     return defer.succeed(client)
 
 
-class _NoNetworkClient(_Client):  # type: ignore # Cannot determine consistent MRO order
+class _NoNetworkClient(_Client):
     """
     Overrides all _Client networking functionality to do nothing.
     """

tahoe-lafs 3399.mypy $ tox -e typechecks
typechecks installed: attrs==20.3.0,Automat==20.2.0,cffi==1.14.4,constantly==15.1.0,cryptography==3.3.1,foolscap @ git+https://github.com/jaraco/foolscap@d70d8ecee926d3693b9fac55aebe13573d4f71c9,hyperlink==20.0.1,idna==2.10,incremental==17.5.0,mypy==0.790,mypy-extensions==0.4.3,mypy-zope @ git+https://github.com/jaraco/mypy-zope@48f912cc3cdb4ee37d4d3783ef466a7dce8bf0a9,pyasn1==0.4.8,pyasn1-modules==0.2.8,pycparser==2.20,PyHamcrest==2.0.2,pyOpenSSL==20.0.1,service-identity==18.1.0,six==1.15.0,Twisted==20.3.0,typed-ast==1.4.1,typing-extensions==3.7.4.3,zope.event==4.5.0,zope.interface==5.2.0,zope.schema==6.0.0
typechecks run-test-pre: PYTHONHASHSEED='3588652114'
typechecks run-test: commands[0] | mypy src
src/allmydata/test/no_network.py:264: error: Cannot determine consistent method resolution order (MRO) for "_NoNetworkClient"
Found 1 error in 1 file (checked 293 source files)
ERROR: InvocationError for command /Users/jaraco/code/public/tahoe-lafs/.tox/typechecks/bin/mypy src (exited with code 1)
_______________________________________________________________ summary _______________________________________________________________
ERROR:   typechecks: commands failed
```

I don't understand why the issue exists on some platforms but not others. Probably we need an issue to track this, since it's not consistent.

Reply:
Addressed in 01147f462.